src/Block/Service/AbstractCategoriesBlockService.php

```diff
@@ -155,8 +157,9 @@ public function configureSettings(OptionsResolver $resolver): void
 
     public function load(BlockInterface $block): void
     {
-        if (is_numeric($block->getSetting('categoryId'))) {
-            $block->setSetting('categoryId', $this->getCategory($block->getSetting('categoryId')));
+        $categoryId = $block->getSetting('categoryId');
+        if (is_int($categoryId) || is_string($categoryId)) {
+            $block->setSetting('categoryId', $this->getCategory($categoryId));
```

Comment:
you might take inspiration on how I solved for media-bundle: https://github.com/sonata-project/SonataMediaBundle/blob/4.x/src/Block/FeatureMediaBlockService.php#L158-L173

Reply:
There is a getCategory doing the job, I was trying to not touch the logic and just solving the phpstan/psalm issue.

Reply:
Yep, but you could end up setting the categoryId to null, because getCategory can return null

Reply:
Not when it's an int or a string

Reply:
Oh I see, it can when nothing is found

Reply:
@jordisala1991 I would say it's not an issue.
If the id is found, the entity is set. If the id is not found, null is set, which is correct since the value wasnt correct